### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by amitsindhwani**
## What changes are you trying to make? (e.g., adding or removing code, refactoring existing code, adding reports)
Lab4- tried different convolutions, kernel(image processing) Box blur, gaussian blur, grayscale images etc). Tried an image through the webcam etc
Lab6- text classification using Neural networks, embeddings, added multiple layers in the model, worked with pre-trained embeddings
## What did you learn from the changes you have made?
How do convolutions work, how can images be resized and processed using different kernels, how do pre-trained embeddings work, and how does supervised CBOW (Continuos Box of Words) work?
How to classify text using Neural networks, how embedding works, how the more frequent text is assigned numerical values (Amazing)
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
None

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
There were a lot of challenges, The model was crashing in Google Colab; it worked after reducing the learning rate, vocab size, etc. The first model worked after multiple hits and trials.

But when I tried to improve the model using pre-trained embeddings, it kept on crashing again and again with the error "Your session crashed after using all available RAM." 
Even tried with a different Gmail account for Google Colab as my earlier account crossed the free limit of GPU, still crashing and it didn't work. Looking for other options that could be helpful

## How were these changes tested?
Google Colab except the last model
## Checklist
- [Yes ] I can confirm that my changes are working as intended

### CODE REVIEW

**Comment by vishnouvina** [comment]
A2 partially complete

**Comment by amitsindhwani** [comment]
Thanks Vishnou for reviewing the pull request, since google colab was crashing, unable to run those! so submitted the code without the results! Please suggest other ways to run!

Is there anything else being missed out?

**Comment by vishnouvina** [comment]
No problem! I marked it as partially complete, and with your A1 submission and the labs you should be good to go to pass this module, so that you can focus on other modules :)

**Comment by amitsindhwani** [comment]
Thanks Vishnou!
